src/modules/species/species.service.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from "@nestjs/common";
+import { PrismaService } from "src/config/prisma/prisma.service";
+import { CreateSpeciesDto } from "./dto";
+
+@Injectable()
+export class SpeciesService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(CreateSpeciesDto: CreateSpeciesDto) {
+    try {
+      this.prisma.species.create({ data: { ...CreateSpeciesDto } });
+    } catch (err) {
+      if (err.code === "P2002") {
```

Comment:
O erro pode não ser um erro do Prisma.

```suggestion
      if (err?.code === "P2002") {
```